rfcs/text/0049/entity.yml

```diff
@@ -0,0 +1,115 @@
+group: 2
+type: group
+short: Fields to describe various types of entities across IT environments.
+description: >
+  The entity fields provide a standardized way to represent and categorize
+  different types of components within an IT environment, including those
+  that don't have dedicated field sets in ECS. An entity represents a discrete,
+  identifiable component that can be described by a set of attributes and
+  maintains its identity over time.
+root: true
+fields:
+
+  - name: entity.id
+    level: core
+    type: keyword
+    short: Unique identifier for the entity.
+    description: >
+      A unique identifier for the entity. When multiple identifiers exist, this should be
+      the most stable and commonly used identifier that: 1) persists across the entity's
+      lifecycle, 2) ensures uniqueness within its scope, 3) is commonly used for queries
+      and correlation, and 4) is readily available in most observations (logs/events).
+      For entities with dedicated field sets (e.g., host, user), this value should match
+      the corresponding *.id field. Alternative identifiers (e.g., ARNs values in AWS, URLs)
+      can be preserved in entity.raw.
+    example: i-04ff5d36be3d6896c, arn:aws:s3:::my-bucket, projects/123456789/locations/us-central1/instances/my-db
+
+  - name: entity.source
+    level: core
+    type: keyword
+    short: Source module or integration that provided the entity data.
+    description: >
+      The module or integration that provided this entity data (similar to event.module).
+      This helps identify the origin of the entity information and aids in troubleshooting
+      or validating data quality.
+    example: aws, azure, gcp, kubernetes, cloudflare
+
+  - name: entity.type
+    level: core
+    type: keyword
+    short: Standardized high-level classification of the entity.
+    description: >
+      A standardized high-level classification of the entity. This provides a normalized way
+      to group similar entities across different providers or systems. There will be an
+      allowed set of values maintained for this field to ensure consistency.
+    example: bucket, database, container, function, queue, host, user, loadbalancer
+
+  - name: entity.sub_type
+    level: extended
+    type: keyword
+    short: The specific type designation for the entity as defined by its provider or system.
+    description: >
+      The specific type designation for the entity as defined by its provider or system. This field
+      provides more granular classification than entity.type. While entity.type provides a normalized
+      classification across different systems, entity.sub_type preserves the provider-specific
+      categorization.
+    example: aws_s3_bucket, gcp_cloud_storage_bucket, azure_blob_container, aws_lambda_function
+
+  - name: entity.name
+    level: core
+    type: keyword
+    multi_fields:
+      - name: text
+        type: text
+    short: The human-readable name of the entity.
```

Comment:
it's not very human readeable in the examples. Do we need additional `entity.title` to capture the human readable title of the entity? as in EC2 instance `maxcold-ec2` with title `EC2 instance for testing`

Reply:
Nice catch in nuance. For now I'll drop the "human readable" part of the description so we can keep things simple. Entity.name is meant to represent the given name of the entity, human-readable or otherwise.

Reply:
https://github.com/elastic/ecs/pull/2461/commits/374ca057f6e1aeff33f5094412a1db0e408750a7